src/requirements/Payment/components/RegisterVaultForm/RegisterVaultForm.tsx

```diff
@@ -49,6 +51,29 @@ const RegisterVaultForm = ({ isDisabled }: Props): JSX.Element => {
     formState: { errors },
   } = useFormContext<RegisterVaultFormType>()
   const chain = useWatch({ name: "chain" })
+  const token = useWatch({ name: "token" })
+
+  const { data: tokenData } = useToken({
+    address: token,
+    chainId: Chains[chain as keyof typeof Chains],
```

Comment:
[We have a `Chain` type for this](https://github.com/guildxyz/guild.xyz/blob/main/src/wagmiConfig/chains.ts#L698) ;)

But I see the root of the problem here & we probably don't need a typecast. We can either add generics to `useWatch` like this:

```ts
const chain = useWatch<RegisterVaultFormType, "chain">({name: "chain"})
```

Or just pass `control` from `useFormContext` to `useWatch`:

```ts
const {
    control,
    register,
    setValue,
    formState: { errors },
  } = useFormContext<RegisterVaultFormType>()
  
const chain = useWatch({ control, name: "token" })
```

I believe either of these solutions will work, so use whichever one you prefer!